src/_pytest/logging.py

```diff
@@ -63,28 +63,36 @@ class ColoredLevelFormatter(logging.Formatter):
 
     def __init__(self, terminalwriter: TerminalWriter, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
+        self._terminalwriter = terminalwriter
         self._original_fmt = self._style._fmt
         self._level_to_fmt_mapping: Dict[int, str] = {}
 
+        for level, color_opts in self.LOGLEVEL_COLOROPTS.items():
+            self.add_color_level(level, *color_opts)
+
+    def add_color_level(self, level, *color_opts):
```

Comment:
can we do anything about the style color options here

every time we add a terminalwriter style into the api we lock this broken api i want to remove more into pytest

how about migrating from pylib color options to colorama options?

Reply:
What do you mean? The TerminalWriter color codes are just ANSI escape codes that shouldn't change too often...

```
@final
class TerminalWriter:
    _esctable = dict(
        black=30,
        red=31,
        green=32,
        yellow=33,
        blue=34,
        purple=35,
        cyan=36,
        white=37,
        Black=40,
        Red=41,
        Green=42,
        Yellow=43,
        Blue=44,
        Purple=45,
        Cyan=46,
        White=47,
        bold=1,
        light=2,
        blink=5,
        invert=7,
    )

```

Reply:
Ok, any timeline on when an alternative will be ready?

Reply:
> Ok, any timeline on when an alternative will be ready?

Unfortunately we don't have a timeline.

I understand @RonnyPfannschmidt's feelings here, the terminal writer markup codes are something we want to move away from, but there are already other APIs which use them, and we don't have a clear clear definition of what we want to use instead. 

Regardless, even if in the future we get around to replace the markup codes with something else,  the same deprecation/upgrade path that we will need to use in other parts of pytest that use markup codes, would be applied here as well.

So I don't think this should be a blocker for the PR.

> how about migrating from pylib color options to colorama options?

That's not a good idea either, because we are then introducing an alternative way to declare markup options which is already deprecated from the start, because I'm sure we don't want to lock the markup definition to `colorama` either.

Reply:
i will have to agree, as i im unable commit to a timeline for anything alternative
its not right to stop this pr from going in just cause its using the api that is actually available